ControlSettings/ControlSettingsTV.cpp

```diff
@@ -0,0 +1,3920 @@
+                printf("Video Format:: NONE\n");
+                strValue = "NONE";
+                break;
+        }
+        return strValue;
+    }
+
+    static const char *getVideoFrameRateTypeToString(tvVideoFrameRate_t frameRate)
+   {
+       const char *strValue = "NONE";
+       switch(frameRate) {
+            case tvVideoFrameRate_24:
+                printf("Video FrameRate: 24\n");
+                strValue = "24";
+                break;
+            case tvVideoFrameRate_25:
+                printf("Video FrameRate: 25\n");
+                strValue = "25";
+                break;
+            case tvVideoFrameRate_30:
+                printf("Video FrameRate: 30\n");
+                strValue = "30";
+                break;
+            case tvVideoFrameRate_50:
+                printf("Video FrameRate: 50\n");
+                strValue = "50";
+                break;
+            case tvVideoFrameRate_60:
+                printf("Video FrameRate: 60\n");
+                strValue = "60";
+                break;
+            case tvVideoFrameRate_23dot98:
+                printf("Video FrameRate: 23.98\n");
+                strValue = "23.98";
+                break;
+            case tvVideoFrameRate_29dot97:
+                printf("Video FrameRate: 29.97\n");
+                strValue = "29.97";
+                break;
+            case tvVideoFrameRate_59dot94:
+                printf("Video FrameRate: 59.94\n");
+                strValue = "59.94";
+                break;
+            default:
+                printf("Video FrameRate: NONE\n");
+                strValue = "NONE";
+                break;
+
+        }
+        return strValue;
+    }
+
+    static JsonArray getSupportedVideoFrameRate(void)
+    {
+        JsonArray supportedFrameRate;
+        tvError_t ret = tvERROR_NONE;
+        char buffer[EXPECTED_FILE_SIZE] = {0};
+        unsigned short totalFrameRates = 0;
+
+        ret = ReadAllModeConfigfile(GENERIC_MODE_CONFIG_FILE,  buffer, "videoframerate");
+        if(ret == tvERROR_NONE){
+            splitstringsfrombuffer(buffer, videoDescBuffer, &totalFrameRates);
+
+            for(int index=0; index<totalFrameRates; index++){
+                supportedFrameRate.Add(videoDescBuffer+(index*VIDEO_DESCRIPTION_NAME_SIZE));
+            }
+        }else{
+            printf("%s : Reading supported video frame rate failed %d", __FUNCTION__, ret);
+        }
+
+        return supportedFrameRate;
+    }
+
+    ControlSettingsTV::ControlSettingsTV(): PluginHost::JSONRPC()
+					    , m_currentHdmiInResoluton (dsVIDEO_PIXELRES_1920x1080)
+                                            , m_videoZoomMode (tvDisplayMode_NORMAL)
+                                            , m_isDisabledHdmiIn4KZoom (false)
+					    , rfc_caller_id()
+    {
+        LOGINFO("Entry\n");
+        ControlSettingsTV::instance = this;
+	CreateHandler({ 2 });
+        Register("getBacklight", &ControlSettingsTV::getBacklight, this);
```

Comment:
why Register and registerMethod

Reply:
In current tvsettings plugin we have different get methods for llama and hisense.
To acheive this, I have followed the same was as "DisplaySettings" implementation.

**For ex :** 
SetBacklight() : We have only one version. so it  registered like below.
registerMethod("setBacklight", &ControlSettingsTV::setBacklight, this);

getBacklight() : We have different get methods for llama and hisense.

Register("getBacklight", &**ControlSettingsTV::getBackligh**t, this);
GetHandler(2)->Register<JsonObject, JsonObject>("getBacklight", &**ControlSettingsTV::getBacklight2**, this);